sentry/src/main/java/io/sentry/MetricsAggregator.java

```diff
@@ -0,0 +1,320 @@
+package io.sentry;
+
+import io.sentry.metrics.CounterMetric;
+import io.sentry.metrics.DistributionMetric;
+import io.sentry.metrics.EncodedMetrics;
+import io.sentry.metrics.GaugeMetric;
+import io.sentry.metrics.IMetricsHub;
+import io.sentry.metrics.Metric;
+import io.sentry.metrics.MetricType;
+import io.sentry.metrics.MetricsHelper;
+import io.sentry.metrics.SetMetric;
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.zip.CRC32;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+@ApiStatus.Internal
+public final class MetricsAggregator implements IMetricsAggregator, Runnable, Closeable {
+
+  @SuppressWarnings({"CharsetObjectCanBeUsed"})
+  private static final Charset UTF8 = Charset.forName("UTF-8");
+
+  private final @NotNull IMetricsHub hub;
+  private final @NotNull ILogger logger;
+
+  private @NotNull TimeProvider timeProvider = System::currentTimeMillis;
```

Comment:
Let's pass it in the constructor, for testability (and remove the `setTimeProvider` method)